### PR TITLE
fix/63: disable gateway if currency is not supported

### LINKED ISSUE #63: Different store currency to location currency shows error, but gateway is still available
### Describe the bug

Having a different store currency to the business location currency shows an error/warning in WP admin, but the gateway is still available in the checkout.

### To reproduce

1. Set your store up using EUR
1. Use a business location using another currency such as USD 
1. See the error appear in the WP admin
> Heads up! Your store currency is EUR but your configured Square business location currency is USD, so payments cannot be processed. Please choose a different business location or change your shop currency.
1. Go to the Checkout block or Checkout shortcode and see the method is still available

**Screenshots**

### Expected behavior

Based on the error message, I'd expect the method to not show

### PR DESCRIPTION
**Pull request by Sidsector9**
### All Submissions:

* [ ] Does your code follow the [WooCommerce Sniffs](https://github.com/woocommerce/woocommerce-sniffs/) variant of WordPress coding standards?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?
* [ ] Will this change require new documentation or changes to existing documentation?

---

### Changes proposed in this Pull Request:

This PR prevents Square gateway from being displayed if the currency is not supported.

Closes #63 .

### Steps to test the changes in this Pull Request:

Please follow the steps from the issue.

Also please check by adding a Gift Card product to the cart.

### Changelog entry

> Fix - Issue with Square payment gateway being shown for unsupported currencies.

### CODE REVIEW

**Review by dkotter — APPROVED** [review]

**Comment by dkotter** [comment]
This can go to UAT once QA passes

**Comment by qasumitbagthariya** [comment]
### QA Update ✅

-------------

I have checked this issue in the fix/63 branch and the related issue is resolved and working as expected.


I tested the following on this branch:
Save card ✅
Change Currency ✅

**Testing Environment** 

<Details>
WordPress: 6.4.3
Theme: Storefront 4.5.4
WooCommerce - 8.5.2
PHP: 8.2.10
Web Server: Nginx 1.20.2
Browser: Chrome 
OS: macOS Ventura 13.3 

</Details>


https://github.com/woocommerce/woocommerce-square/assets/67687255/729616da-e075-48a7-83ee-0fe9905c0b7c



Steps to Test- As mentioned in the PR description.
Test Results - It is working as expected.
Functional Demo / Screencast - 
Special Notes -  Ready for UAT
Testing Document status:
Cases related to this Issue/PR are added to the Critical Flow Wiki pages: 

- [x] Yes
- [ ] Not Required/Applicable for this PR

File: changelog.txt [hunk 1]
```diff
@@ -1,5 +1,8 @@
 *** WooCommerce Square Changelog ***
 
+= 4.5.1 - TBD =
```
Comment (jeffpaul) [inline changelog.txt]: @Sidsector9 note that you can stop applying updates in the changelog.txt and readme.txt files for the changelog entries as we can now apply those all at once during the extension release process.

**Review by qasumitbagthariya — APPROVED** [review]

**Comment by qasumitbagthariya** [comment]
Regression / Smoke Test Report  ✅ 
-------

Tested with Archive File created via "php woorelease.phar build repo_URL" (Composer version 2.5.5, npm version 8.19.4, node version 16.20.0)

Status- Working expected with Plugin Archive/Zip file same as fix specific branch.

Next Step- Ready to Merge 🚀

Status update:
cc @vikrampm1

**Comment by dkotter** [comment]
Note that E2E tests are failing on this PR (see the cancelled run: https://github.com/woocommerce/woocommerce-square/actions/runs/8069258422/job/22043925467?pr=67). We'll need to address this before proceeding with getting this merged/released

**Comment by Sidsector9** [comment]
@dkotter the tests are passing now 👍

**Review by dkotter — APPROVED** [review]

**Pull request merged**